### PR TITLE
GPC-349: Add infrastructure setup in

### PR DESCRIPTION
**Pull request by oskarwilliams378**

### CODE REVIEW

File: ci/update_securityhub_suppressions/suppressions.yml [hunk 1]
```diff
@@ -302,6 +304,8 @@ Suppressions:
         - arn:aws:lambda:eu-west-2:776473272850:function:dev-get-event
         - arn:aws:lambda:eu-west-2:776473272850:function:dev-get-events
         - arn:aws:lambda:eu-west-2:776473272850:function:dev-delete-event
+        - arn:aws:lambda:eu-west-2:776473272850:function:dev-gro-ingestion-lambda-function-publish-event
+        - arn:aws:lambda:eu-west-2:776473272850:function:demo-gro-ingestion-lambda-function-publish-event
```
Comment (ethanmills) [inline ci/update_securityhub_suppressions/suppressions.yml:308]: I don't think this note applies to these functions?

File: terraform/modules/gro_ingestion_service/publish_event.tf [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  handler = "index.handler"
+  runtime = local.lambda_runtime
+  role    = aws_iam_role.publish_event_lambda.arn
+  timeout = 10
+
+  source_code_hash = data.archive_file.lambda_function_source.output_sha
+
+  environment {
+    variables = {
+      "FUNCTION_NAME" = "publishEvent"
```
Comment (ethanmills) [inline terraform/modules/gro_ingestion_service/publish_event.tf:29]: Is this used?

Comment (oskarwilliams378) [reply terraform/modules/gro_ingestion_service/publish_event.tf:29]: This is what lets the code get triggered via the index

File: ci/update_securityhub_suppressions/suppressions.yml (hunk 1 above)
Comment (oskarwilliams378) [reply ci/update_securityhub_suppressions/suppressions.yml:308]: I've removed test, but it does. The lambdas are proxies that only access our system over the public internet

File: terraform/demo/main.tf [hunk 3]
```diff
@@ -109,4 +109,9 @@ module "gro_ingestion_service" {
 
   environment = local.env
   region      = data.aws_region.current.name
+
+  gdx_url                 = module.data_share_service.gdx_url
+  auth_url                = module.data_share_service.token_auth_url
+  publisher_client_id     = module.data_share_service.gro_ingestion_client_id
+  publisher_client_secret = module.data_share_service.gro_ingestion_client_secret
```
Comment (olevett) [inline terraform/demo/main.tf:116]: Is this the raw credential passing around/will it end up tfstate? If so, is there any sensible way we can use SSM/inject the secret without knowing it as much?

**Review by olevett — APPROVED** [review]

File: terraform/demo/main.tf (hunk 3 above)
Comment (oskarwilliams378) [reply terraform/demo/main.tf:116]: It will end up in tfstate, I can mark it as sensitive, but this is the same as the current setup for our consumer.

**Pull request merged**